File: src/cpp_pkg/src/publisher_node.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

using namespace std::chrono_literals;

class PublisherNode : public rclcpp::Node
{
    public:
        PublisherNode() : Node("publisher")
        {
            publisher_ = this->create_publisher<example_interfaces::msg::String>("simple_topic", 10);
            timer_ = this->create_wall_timer(500ms, std::bind(&PublisherNode::publish_example, this));
            RCLCPP_INFO(this->get_logger(), "Publisher has been started ...");
        }
        
    private:
        void publish_example()
        {
            auto msg = example_interfaces::msg::String();
            msg.data = "Simple publisher";
            publisher_->publish(msg);
        } 
    
        rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
        rclcpp::TimerBase::SharedPtr timer_;
}; 

int main(int argc, char **argv){

    rclcpp::init(argc, argv);
    auto node = std::make_shared<PublisherNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/py_pkg/py_pkg/publisher_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

class PublisherNode(Node):

    def __init__(self):
        super().__init__("publisher")
        self.publisher_ = self.create_publisher(String, "simple_topic", 10)
        self.timer_ = self.create_timer(0.5, self.publish_example)
        self.get_logger().info("Publisher has been started ...")

    def publish_example(self):
        msg = String()
        msg.data = "Simple publisher"
        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = PublisherNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()